dell/src/main/java/org/apache/iceberg/dell/EcsOutputFile.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.dell;
+
+import com.emc.object.s3.S3Client;
+import org.apache.iceberg.exceptions.AlreadyExistsException;
+import org.apache.iceberg.io.InputFile;
+import org.apache.iceberg.io.OutputFile;
+import org.apache.iceberg.io.PositionOutputStream;
+
+public class EcsOutputFile implements OutputFile {
+
+  private final S3Client client;
+  private final String location;
+  private final EcsURI uri;
+
+  public EcsOutputFile(S3Client client, String location) {
+    this.client = client;
+    this.location = location;
+    this.uri = LocationUtils.checkAndParseLocation(location);
+  }
+
+  /**
+   * Check object existence and then create a {@link PositionOutputStream}
+   *
+   * @return Output stream of object
+   */
+  @Override
+  public PositionOutputStream create() {
+    if (!toInputFile().exists()) {
+      return createOrOverwrite();
+    } else {
+      throw new AlreadyExistsException("The object is existed, %s", uri);
+    }
+  }
+
+  @Override
+  public PositionOutputStream createOrOverwrite() {
+    // use built-in 1 KiB byte buffer
+    return new EcsAppendOutputStream(client, uri, new byte[1_000]);
```

Comment:
why not directly use `ByteBuffer.allocate`? Then you don't need to convert it back to byte buffer in the constructor.  Also would it be more beneficial to allocate 1024 vs 1000?

Reply:
And I think we prefer to pass ByteBuffer as argument instead of direct byte array. So could you at least move the `ByteBuffer.wrap` to this level, or the constructor don't really need to have this argument since you always create in a new byte buffer of fixed length, so you can directly create it in the constructor instead.

Reply:
And could you make your buffer size a private static constant for readability, thanks!

Reply:
Done. I use a factory method.